### PR TITLE
kitty-themes: init at unstable-2022-01-01

### LINKED ISSUE #153504: add kitty-themes
**kitty-themes**
_Themes for the kitty terminal emulator_

**Metadata**

* homepage URL: https://github.com/kovidgoyal/kitty-themes
* source URL: https://github.com/kovidgoyal/kitty-themes
* license: gpl3
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by NelsonJeppesen**
###### Motivation for this change
Closes https://github.com/NixOS/nixpkgs/issues/153504

Add kitty-themes to the Nix store. Once there, I can use them with `home-manager`. Once merged, I plan to add a themes option to home-managers kitty module

###### Things done
This is very basic; copy all the `*.conf` theme files and manifest to Nix store from GitHub
```
❯ find /nix/store/0r0v51dyd905n7n2drzxhg3xk0jdyjkf-kitty-themes-unstable-2022-02-03 | head -n 5
/nix/store/0r0v51dyd905n7n2drzxhg3xk0jdyjkf-kitty-themes-unstable-2022-02-03
/nix/store/0r0v51dyd905n7n2drzxhg3xk0jdyjkf-kitty-themes-unstable-2022-02-03/themes.json
/nix/store/0r0v51dyd905n7n2drzxhg3xk0jdyjkf-kitty-themes-unstable-2022-02-03/themes
/nix/store/0r0v51dyd905n7n2drzxhg3xk0jdyjkf-kitty-themes-unstable-2022-02-03/themes/1984_dark.conf
/nix/store/0r0v51dyd905n7n2drzxhg3xk0jdyjkf-kitty-themes-unstable-2022-02-03/themes/1984_light.conf
...
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/kitty-themes/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    mv themes/*.conf $out
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/kovidgoyal/kitty-themes";
+    description = "Themes for the kitty terminal emulator";
+    license = licenses.gpl3; # from debian/copyright
```
Comment (Luflosi) [inline pkgs/applications/terminal-emulators/kitty-themes/default.nix:24]:
```suggestion
    license = licenses.gpl3Only;
```

File: pkgs/applications/terminal-emulators/kitty-themes/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    platforms = platforms.darwin ++ platforms.linux;
+    maintainers = with maintainers; [ tex rvolosatovs Luflosi ];
```
Comment (Luflosi) [inline pkgs/applications/terminal-emulators/kitty-themes/default.nix:26]: You should add yourself to this list and remove the three others.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -1102,6 +1102,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Cocoa CoreGraphics Foundation IOKit Kernel OpenGL;
   };
 
+  kitty-themes  = callPackage ../applications/terminal-emulators/kitty-themes  {};
```
Comment (Luflosi) [inline pkgs/top-level/all-packages.nix:1105]: I think `terminal-emulators` is not the right category here since this is not a terminal emulator but a collection of themes. `/pkgs/data/themes/` seems to also not fit because the `coding-conventions` document says to use that directory "If it’s a theme for a desktop environment, a  window manager or a display manager", which is not the case for kitty-themes. So maybe just put it in `/pkgs/misc/`?

File: pkgs/applications/terminal-emulators/kitty-themes/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "kitty-themes";
+  version = "2021-01-04";
+
+  src = fetchFromGitHub {
+    owner = "kovidgoyal";
+    repo = pname;
+    rev = "fca3335489bdbab4cce150cb440d3559ff5400e2";
```
Comment (Luflosi) [inline pkgs/applications/terminal-emulators/kitty-themes/default.nix:12]: Why are you using such an old commit? https://github.com/kovidgoyal/kitty-themes/commit/fca3335489bdbab4cce150cb440d3559ff5400e2 is from 2019. Use the latest one instead, which is https://github.com/kovidgoyal/kitty-themes/commit/76e0c1151ea2d48a6c9c3842095c9781051e5b88.

File: pkgs/applications/terminal-emulators/kitty-themes/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+  fetchFromGitHub, lib, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "kitty-themes";
+  version = "2021-01-04";
```
Comment (Luflosi) [inline pkgs/applications/terminal-emulators/kitty-themes/default.nix:7]: This date should reflect the date when the commit below was made. In this case it should be 2022-01-01.

**Review by Luflosi — CHANGES_REQUESTED** [review]

**Comment by NelsonJeppesen** [comment]
@Luflosi Thank you for your feedback. I've made the changes requested. Add myself to maintainers and ran the check script 

```
❯ ./maintainers/scripts/check-maintainer-github-handles.sh
Academic tradition requires you to cite works you base your article on.
If you use programs that use GNU Parallel to process data for an article in a
scientific publication, please cite:

  Tange, O. (2021, December 22). GNU Parallel 20211222 ('Støjberg').
  Zenodo. https://doi.org/10.5281/zenodo.5797028

This helps funding further development; AND IT WON'T COST YOU A CENT.
If you pay 10000 EUR you should feel free to use GNU Parallel without citing.

More about funding GNU Parallel and the citation notice:
https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice

To silence this citation notice: run 'parallel --citation' once.

OK!				NelsonJeppesen
```

File: pkgs/misc/kitty-themes/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{
+  fetchFromGitHub, lib, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "kitty-themes";
+  version = "2021-01-01";
```
Comment (Luflosi) [inline pkgs/misc/kitty-themes/default.nix:7]:
```suggestion
  version = "2022-01-01";
```
The commit below is from January 2022 and not January 2021. Please also update the title of the pull request and the commit message.

Comment (Luflosi) [reply pkgs/misc/kitty-themes/default.nix:7]: I just saw, that the commit message is already correct.

Comment (NelsonJeppesen) [reply pkgs/misc/kitty-themes/default.nix]: @Luflosi my apologies; updated

File: pkgs/misc/kitty-themes/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    mkdir -p $out
+    mv themes/*.conf $out
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/kovidgoyal/kitty-themes";
+    description = "Themes for the kitty terminal emulator";
+    license = licenses.gpl3Only;
+    platforms = platforms.darwin ++ platforms.linux;
```
Comment (Luflosi) [inline pkgs/misc/kitty-themes/default.nix:25]: One last suggestion from me:
```suggestion
    platforms = platforms.all;
```
Since this package contains no executable code, it should build on every architecture.

**Comment by NelsonJeppesen** [comment]
@Luflosi that's a very good point. I've updated to your suggestion

Thank you for the help learning nixpkgs

**Review by Luflosi — APPROVED** [review]
LGTM.

**Comment by Luflosi** [comment]
Oh, wait, I just found another thing. The version should be `unstable-2022-01-01` instead of `2022-01-01` because it's a commit and not a real released version.

**Comment by NelsonJeppesen** [comment]
> Oh, wait, I just found another thing. The version should be `unstable-2022-01-01` instead of `2022-01-01` because it's a commit and not a real released version.

Roger - updated commit, version and pr

**Review by Luflosi — APPROVED** [review]

**Comment by NelsonJeppesen** [comment]
Bumped version to last commit on the 3rd; added theme manifest

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -8444,6 +8444,12 @@
     githubId = 3747396;
     name = "Nathan Isom";
   };
+  NelsonJeppesen = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8447]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/misc/kitty-themes/default.nix [hunk 9]
```diff
@@ -0,0 +1,33 @@
+{
+  fetchFromGitHub, lib, stdenv
+}:
```
Comment (SuperSandro2000) [inline pkgs/misc/kitty-themes/default.nix:3]:
```suggestion
{ fetchFromGitHub, lib, stdenv }:
```

File: pkgs/misc/kitty-themes/default.nix [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  installPhase = ''
+    # copy theme manifest
+    mkdir -p $out
+    mv themes.json $out
+
+    # copy themes
+    mkdir -p $out/themes
+    mv themes/*.conf $out/themes
```
Comment (SuperSandro2000) [inline pkgs/misc/kitty-themes/default.nix:23]:
```suggestion
    mkdir -p $out/themes
    mv themes.json $out
    mv themes/*.conf $out/themes
```

File: pkgs/top-level/all-packages.nix [hunk 11]
```diff
@@ -1104,6 +1104,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Cocoa CoreGraphics Foundation IOKit Kernel OpenGL;
   };
 
+  kitty-themes  = callPackage ../misc/kitty-themes  {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1107]:
```suggestion
  kitty-themes  = callPackage ../misc/kitty-themes {};
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by NelsonJeppesen** [comment]
Thank you @SuperSandro2000 for the review and feedback

I believe I had resolved the issues you have identified
